R/checks.R

```diff
@@ -252,14 +252,41 @@ check_match_type <- function(x, y, by, match_type, verbose) {
 
   # Error if user chooses "1" but actually "m" ----
   if (match_type_error) {
+
     msg     <- "match type inconsistency"
     hint    <-
-      "you could use `return_report = TRUE` in `joyn::is_id()`
-    to see where the problem is"
+      "refer to the duplicate counts in the table(s) above
+       to identify where the issue occurred"
     joyn_msg("err")
+
+    display_id_x <- display_id_y <- NULL
+
+    # if x is not id (i.e., if xm = TRUE and user chooses 1:..)
+    if (x_m & tx == 1) {
+      display_id_x <- is_id(x, by, return_report = TRUE, verbose = FALSE) |>
+        fsubset(copies > 1)
+    }
+
+    # if y is not id (i.e., if ym = TRUE and user chooses ...:1)
+    if (y_m & ty == 1) {
```

Comment:
Same problem as above. This is why, in the example below you get that `y1` is not uniquely identified, when it actually is 

```r
x1 = data.frame(
      id = c(1L, 1L, 2L, 3L, NA_integer_),
      t  = c(1L, 2L, 1L, 2L, NA_integer_),
      x  = 11:15)
y1 = data.frame(id = 1:2,
               y  = c(11L, 15L))
joyn:::check_match_type(x = x1, y=y1, by="id", match_type = "1:1")
```